### ISSUE TITLE
IE11 does not work if `<script type="module">` tag appears before `<script nomodule>` tag

### ISSUE THREAD

**Issue by jussih** [ionic-team/stencil#2090]
**Stencil version:**

```
 @stencil/core@1.8.1
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**

If an event listener function is named the same as a custom event emitted inside a stencil component, the polyfilled IE & Edge www bundle generated with `npm run build` does not work. The content of the web component DOM element is empty. No errors appear in the console.

The generated ESM bundle works with Firefox.

**Expected behavior:**

The generated polyfilled bundle works the same as the ESM bundle in Firefox, ie. the content of the web component is rendered.

**Steps to reproduce:**

**Related code:**

This triggers the problem:
```tsx
  @Listen("menuItemExpandToggled")
  menuItemExpandToggled(event: CustomEvent<MenuItem>): void {
    this.menuItems = this.menuItems.map(item =>
      item.route.id === event.detail.route.id
        ? { ...item, isExpanded: !item.isExpanded }
        : item
    );
  }
```
This version of the code works as expected:
```tsx
  @Listen("menuItemExpandToggled")
  fooBar(event: CustomEvent<MenuItem>): void {
    this.menuItems = this.menuItems.map(item =>
      item.route.id === event.detail.route.id
        ? { ...item, isExpanded: !item.isExpanded }
        : item
    );
  }
```

**Other information:**

I suspect there is some name collision in the polyfilled js bundle.

The bug is not a showstopper since with good naming it does not manifest, but it took quite a while to debug with git bisect why the component was not working in IE, especially as there was no console errors hinting what was wrong.

**Comment by jussih** [comment]
Does not seem to be a naming clash after all. The length in characters of the handler function name matters.

This fails:
```
  @Listen("menuItemExpandToggled")
  abcdefghijklmnopqr(event: CustomEvent<MenuItem>): void {
```

This works:
```
  @Listen("menuItemExpandToggled")
  abcdefghijklmnopq(event: CustomEvent<MenuItem>): void {
```

If I go above 17 characters in the function name, the IE bundle stops working.

**Comment by jussih** [comment]
I ran into the same problem again and had to investigate more.

Turns out that removing characters from the function name resulted in Stencil build inlining the `<script type="module">` contents and moving the tag to the end of the body in index.html. When the module ens up being larger, the script tag stays put where it is in `src/index.html` - at least in my case that's just above the `<script nomodule>` tag containing the IE script bundle.

For whatever mysterious reason, the IE bundle does not work if the nomodule script tag comes after the type="module" script. Flipping the script tags in index.html so the nomodule script tag is first fixes the problem.

I guess this is probably not a Stencil issue. The stars just are not aligned correctly for IE. I'll close the issue here.
